server/src/main/java/org/jboss/as/server/ServerEnvironment.java

```diff
@@ -332,47 +338,54 @@ public ServerEnvironment(final String hostControllerName, final Properties props
+                             final RunningMode initialRunningMode, ProductConfig productConfig, long startTime, String gitRepository,
+                             String gitBranch, String gitAuthConfiguration, String yaml) {
         this(hostControllerName, props, env, serverConfig, configInteractionPolicy, launchType, initialRunningMode, productConfig,
-                startTime, false, gitRepository, gitBranch, gitAuthConfiguration);
+                startTime, false, gitRepository, gitBranch, gitAuthConfiguration, yaml);
     }
 
     @Deprecated
     public ServerEnvironment(final String hostControllerName, final Properties props, final Map<String, String> env, final String serverConfig,
                              final ConfigurationFile.InteractionPolicy configInteractionPolicy, final LaunchType launchType,
                              final RunningMode initialRunningMode, ProductConfig productConfig, long startTime, boolean startSuspended,
-                             String gitRepository, String gitBranch, String gitAuthConfiguration) {
+                             String gitRepository, String gitBranch, String gitAuthConfiguration, String yaml) {
         this(hostControllerName, props, env, serverConfig, configInteractionPolicy, launchType, initialRunningMode,
-                productConfig, startTime, startSuspended, false, gitRepository, gitBranch, gitAuthConfiguration);
+                productConfig, startTime, startSuspended, false, gitRepository, gitBranch, gitAuthConfiguration, yaml);
     }
 
     public ServerEnvironment(final String hostControllerName, final Properties props, final Map<String, String> env, final String serverConfig,
-                             final ConfigurationFile.InteractionPolicy configInteractionPolicy, final LaunchType launchType,
+                             final ConfigurationFile.InteractionPolicy configurationInteractionPolicy, final LaunchType launchType,
                              final RunningMode initialRunningMode, ProductConfig productConfig, long startTime, boolean startSuspended,
-                             final boolean startGracefully, final String gitRepository, final String gitBranch, final String gitAuthConfiguration) {
+                             final boolean startGracefully, final String gitRepository, final String gitBranch, final String gitAuthConfiguration,
+                             final String yaml) {
         assert props != null;
-
+        ConfigurationFile.InteractionPolicy configInteractionPolicy = configurationInteractionPolicy;
         this.startSuspended = startSuspended;
         this.startGracefully = startGracefully;
 
         this.launchType = launchType;
         this.standalone = launchType != LaunchType.DOMAIN;
 
         this.initialRunningMode = initialRunningMode == null ? RunningMode.NORMAL : initialRunningMode;
+        Path[] yamlFiles = findYamlFiles(yaml);
```

Comment:
what is `yaml`? Is the the path to YAML files? or YAML content?

Reply:
path to the yaml file